### PR TITLE
Added support for docker (+docker-compose)

### PR DESCRIPTION
**Pull request by bilak**
Fixes gh-3

### CODE REVIEW

File: api-gateway/src/main/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM java:8
```
Comment (kakawait) [inline api-gateway/src/main/docker/Dockerfile]: Maybe switch to `java:8-jre-alpine`?

File: api-gateway/src/main/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+ADD api-gateway.jar app.jar
```
Comment (kakawait) [inline api-gateway/src/main/docker/Dockerfile]: You can use `filtering` configuration from plugin `docker-maven-plugin` to be able to 

```
ADD @project.build.finalName@.jar app.jar
```

**Comment by kakawait** [comment]
@bilak I already think we should create a maven `profile` to be able to build whole service with their images

```
mvn clean install -Pdocker (or -Pcontainer whatever)
```

File: api-gateway/src/main/docker/Dockerfile (hunk 2 above)
Comment (kakawait) [inline api-gateway/src/main/docker/Dockerfile]: Is not really mandatory. Just by using `filtering` we can have generic `Dockerfile`

**Comment by kakawait** [comment]
@bilak If you have any trouble with maven configuration (about `profile` and multi-module) I can help you I already done such configurations for other project

**Comment by bilak** [comment]
Btw I've only tested if I can get the token. Didn't tested if dummy service works with this token, because I don't work with JWT's so that could be tested by you probably. There are plenty options which we can work on, maybe you can create some checklist.

**Comment by bilak** [comment]
@kakawait maybe create another issue to don't mess with a lot of commits.

**Comment by kakawait** [comment]
@bilak yes please only support `JWT` (default implementation) for this *PR* and open issue to add other possible options/configurations like without `JWT`

**Comment by kakawait** [comment]
@bilak do you continue working on PR? Else I can polish current and merge it asap

**Comment by bilak** [comment]
@kakawait sorry currently I don't have time for this.

**Comment by kakawait** [comment]
@bilak I continue work on #14 
Thank you for PR, I used a lot of part for new PR #14 

I close this PR in favor to #14

**Pull request closed without merging**